File: src/main/java/org/noxet/noxetserver/playerdata/types/PDTString.java
package org.noxet.noxetserver.playerdata.types;

import org.bukkit.configuration.file.YamlConfiguration;
import org.noxet.noxetserver.playerdata.PlayerDataType;

public class PDTString implements PlayerDataType<String> {
    @Override
    public String getEmptyValue() {
        return null;
    }

    @Override
    public String getValue(YamlConfiguration config, String key) {
        return config.getString(key);
    }
}

File: src/main/java/org/noxet/noxetserver/playerdata/types/PDTString.scala
package org.noxet.noxetserver.playerdata.types

import org.bukkit.configuration.file.YamlConfiguration
import org.noxet.noxetserver.playerdata.PlayerDataType

class PDTString extends PlayerDataType[String]:
  override def getEmptyValue: String = null

  override def getValue(config: YamlConfiguration, key: String): String =
    config.getString(key)